Detectors/TPC/base/include/TPCBase/CalArray.h

```diff
@@ -165,6 +198,71 @@ inline const CalArray<T>& CalArray<T>::operator+= (const CalArray& other)
 
 //______________________________________________________________________________
 template <class T>
+inline const CalArray<T>& CalArray<T>::operator-= (const CalArray& other)
+{
+  if ( !((mPadSubset == other.mPadSubset) && (mPadSubsetNumber == other.mPadSubsetNumber) ) ){
+    LOG(ERROR) << "You are trying to operate on incompatible objects: Pad subset type and number must be the same on both objects"
+               << FairLogger::endl;
+    return *this;
+  }
+  for (size_t i=0; i<mData.size(); ++i) {
+    mData[i] -= other.getValue(i);
+  }
+  return *this;
+}
+
+//______________________________________________________________________________
+template <class t>
+inline const CalArray<t>& CalArray<t>::operator*= (const CalArray& other)
+{
```

Comment:
I think we should annotate the `const CalArray& other` part as `const CalArray<T>& other` to make sure it is the same typed array that we are copying.